OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseOutlineFragment.java

```diff
@@ -398,6 +478,54 @@ public void viewDownloadsStatus() {
         }
     }
 
+    /**
+     * Initialized dates info banner on CourseOutlineFragment
+     *
+     * @param courseBannerInfo object of course deadline info
+     */
+    private void initDatesBanner(CourseBannerInfoModel courseBannerInfo) {
+        LayoutCourseDatesBannerBinding bannerViewBinding = DataBindingUtil.inflate(getLayoutInflater(), R.layout.layout_course_dates_banner, listView, false);
+        if (courseBannerInfo != null && !isVideoMode && isOnCourseOutline && !courseBannerInfo.getHasEnded()) {
+            String buttonText = "";
+            // Currently we are only handling RESET_DATES case,
+            // UPGRADE_TO_GRADED & UPGRADE_TO_RESET will be enable once we are allowed to do payment through mobile
+            switch (courseBannerInfo.getDatesBannerInfo().getCourseBannerType()) {
+                case UPGRADE_TO_GRADED:
+                    bannerViewBinding.bannerInfo.setText(getText(R.string.course_dates_banner_upgrade_to_graded));
+//                    buttonText = getContextOrThrow().getString(R.string.course_dates_banner_upgrade_to_graded_button);
+                    break;
+                case UPGRADE_TO_RESET:
+                    bannerViewBinding.bannerInfo.setText(getText(R.string.course_dates_banner_upgrade_to_reset));
+//                    buttonText = getContextOrThrow().getString(R.string.course_dates_banner_upgrade_to_reset_button);
+                    break;
+                case RESET_DATES:
+                    bannerViewBinding.bannerInfo.setText(getText(R.string.course_dates_banner_reset_date));
+                    buttonText = getContextOrThrow().getString(R.string.course_dates_banner_reset_date_button);
+                    break;
+                case INFO_BANNER:
+                    bannerViewBinding.bannerInfo.setText(getText(R.string.course_dates_info_banner));
+                    break;
+                case BLANK:
+                    break;
+            }
+
+            if (listView.getHeaderViewsCount() == 0 && !TextUtils.isEmpty(bannerViewBinding.bannerInfo.getText())) {
+                if (!TextUtils.isEmpty(buttonText)) {
+                    bannerViewBinding.btnShiftDates.setText(buttonText);
+                    bannerViewBinding.btnShiftDates.setVisibility(View.VISIBLE);
+                    bannerViewBinding.btnShiftDates.setOnClickListener(v -> {
+                        resetCourseDatesBanner(true);
```

Comment:
no need to pass the boolean as always show progress.

Reply:
Removed